File: server_C_sharp_client_Java_final_protoV3/mpp_c_app1-SERVER_c/FestivalServer/StartProtoServer.cs

using System;
using System.Collections.Generic;
using System.Configuration;
using System.Net.Sockets;
using System.Threading;
using FestivalNetworking;
using FestivalNetworking.ServerTemplate;
using FestivalPersistence;
using FestivalService;


namespace FestivalServer
{
    class StartProtoServer
    {
        private static int DEFAULT_PORT = 55556;
        private static String DEFAULT_IP = "127.0.0.1";

        static void Main(string[] args)
        {
            Console.WriteLine("Reading properties from app.config ... ");
            int port = DEFAULT_PORT;
            String ip = DEFAULT_IP;
            String portS = ConfigurationManager.AppSettings["port"];
            if (portS == null)
            {
                Console.WriteLine("Port property not set. Using default value " + DEFAULT_PORT);
                
            }
            else
            {
                bool result = Int32.TryParse(portS, out port);
                if (!result)
                {
                    Console.WriteLine("Port propery not a number. Using default value " + DEFAULT_PORT);
                    port = DEFAULT_PORT;
                    Console.WriteLine("Portul " + port);
                }
            }

            String ipS = ConfigurationManager.AppSettings["ip"];

            if (ipS == null)
            {
                Console.WriteLine("Port property not set. Using default value " + DEFAULT_IP);
            }
            //Console.WriteLine("Configuration Settings for database {0}", GetConnectionStringByName("MPPCurse"));
            //IDictionary<String, String> props = new SortedList<string, string>();
            //props.Add("ConnectionString", GetConnectionStringByName("MPPCurse"));
            DBRepoPersoana persoanaRepo =
                new DBRepoPersoana(
                    "Data Source=C:/Users/andre/Documents/facultate/an2/MAP/mpp_c_app1-SERVER_c/festival.db");

            DBRepoArtist artistRepo =
                new DBRepoArtist(
                    "Data Source=C:/Users/andre/Documents/facultate/an2/MAP/mpp_c_app1-SERVER_c/festival.db");

            DBRepoBilet biletRepo =
                new DBRepoBilet(
                    "Data Source=C:/Users/andre/Documents/facultate/an2/MAP/mpp_c_app1-SERVER_c/festival.db");

            DBRepoSpectacol spectacolRepo =
                new DBRepoSpectacol(
                    "Data Source=C:/Users/andre/Documents/facultate/an2/MAP/mpp_c_app1-SERVER_c/festival.db");

            DBRepoAngajat angajatRepo =
                new DBRepoAngajat(
                    "Data Source=C:/Users/andre/Documents/facultate/an2/MAP/mpp_c_app1-SERVER_c/festival.db");
            DBRepoClient clientRepo =
                new DBRepoClient(
                    "Data Source=C:/Users/andre/Documents/facultate/an2/MAP/mpp_c_app1-SERVER_c/festival.db");
            AuthenticationService autentificare =
                new AuthenticationService(
                    "Data Source=C:/Users/andre/Documents/facultate/an2/MAP/mpp_c_app1-SERVER_c/festival.db");

            // IUserRepository userRepo=new UserRepositoryMock();
            // IMessageRepository messageRepository=new MessageRepositoryMock();
            IService serviceImpl = new Service(biletRepo,artistRepo,spectacolRepo,angajatRepo,clientRepo,persoanaRepo,autentificare);

            Console.WriteLine("Starting server on IP {0} and port {1}", ip, port);
            //SerialServer serialServer = new SerialServer(ip, port, service);
            //serialServer.Start();
            ProtoV3ChatServer scs = new ProtoV3ChatServer(ip, port, serviceImpl);
            scs.Start();
            Console.WriteLine("Server started ... ");
            Console.ReadLine();
        }

        static string GetConnectionStringByName(string name)
        {
            string returnValue = null;

            ConnectionStringSettings settings = ConfigurationManager.ConnectionStrings[name];
            if (settings != null)
                returnValue = settings.ConnectionString;

            return returnValue;
        }
        
        public class SerialServer: ConcurrentServer 
        {
            private IService server;
            private ClientWorker worker;
            public SerialServer(string host, int port, IService server) : base(host, port)
            {
                this.server = server;
                Console.WriteLine("SerialChatServer...");
            }
            protected override Thread createWorker(TcpClient client)
            {
                worker = new ClientWorker(server, client);
                return new Thread(new ThreadStart(worker.run));
            }
        }
        
        public class ProtoV3ChatServer : ConcurrentServer
        {
            private IService server;
            private ProtoWorker worker;
            public ProtoV3ChatServer(string host, int port, IService server)
                : base(host, port)
            {
                this.server = server;
                Console.WriteLine("ProtoChatServer...");
            }
            protected override Thread createWorker(TcpClient client)
            {
                worker = new ProtoWorker(server, client);
                return new Thread(new ThreadStart(worker.run));
            }
        }

    }
    
}/**/

File: server_C_sharp_client_Java_final_protoV3/JavaSocketsFestival-CLIENT_java/Java11ChatSockets/ChatServer/src/main/java/StartProtoServer.java


import festival.network.utils.AbstractServer;
import festival.network.utils.ChatProtobuffConcurrentServer;
import festival.network.utils.ServerException;
import festival.persistence.repository.*;
import festival.server.Service;
import festival.services.ChatException;
import festival.services.IChatServices;

import java.io.IOException;
import java.util.Properties;


public class StartProtoServer {
    private static int defaultPort=55555;
    public static void main(String[] args) {


        Properties serverProps=new Properties();
        try {
            serverProps.load(StartRpcServer.class.getResourceAsStream("/chatserver.properties"));
            System.out.println("Server properties set. ");
            serverProps.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find chatserver.properties "+e);
            return;
        }
        DBRepoArtist ArtistRepo=new DBRepoArtist(serverProps);
        DBRepoSpectacol SpectacolRepo=new DBRepoSpectacol(serverProps);
        DBRepoBilet BiletRepo=new DBRepoBilet(serverProps);
        DBRepoAngajat AngajatRepo=new DBRepoAngajat(serverProps);
        DBRepoClient ClientRepo=new DBRepoClient(serverProps);
        DBRepoPersoana PersoanaRepo=new DBRepoPersoana(serverProps);



        IChatServices srv=new Service(BiletRepo,ArtistRepo,SpectacolRepo,AngajatRepo,ClientRepo,PersoanaRepo);

        int chatServerPort=defaultPort;
        try {
            chatServerPort = Integer.parseInt(serverProps.getProperty("chat.server.port"));
        }catch (NumberFormatException nef){
            System.err.println("Wrong  Port Number"+nef.getMessage());
            System.err.println("Using default port "+defaultPort);
        }
        System.out.println("Starting server on port: "+chatServerPort);
        AbstractServer server = new ChatProtobuffConcurrentServer(chatServerPort, srv);
        try {
            server.start();
        } catch (ServerException e) {
            throw new RuntimeException(e);
        }


    }
}
